### PR TITLE
Ignore _about.py in pytest coverage

### PR DESCRIPTION
**Pull request by purva-thakre**
Related to two of the items in #2365 

1. mitiq/_about.py
2. mitiq/pec/representations/optimal.py

### CODE REVIEW

**Comment by purva-thakre** [comment]
@cosenal This one's ready to be merged as well. Unless we don't want to ignore this file from the pytest coverage report.

**Review by natestemen — COMMENTED** [review]
As far as I understand, there are no existing tests for the `mitiq.about()` function[^1], so ignoring `mitiq/_about.py` will make coverage go up. Should there be tests for this function, however?

[^1]: `mitiq.about` is called in some of the notebooks, so that's an implicit test that it works, but no unit tests.

**Comment by natestemen** [comment]
> I was confused if this file needs to have unit tests.

A test that ensures `mitiq.about()` returns a string seems reasonable. I wouldn't want to test much more than that, however. Definitely low priority.

**Comment by purva-thakre** [comment]
> A test that ensures mitiq.about() returns a string seems reasonable.

Output of `mitiq.about()` is `NoneType` not `str`.

**Comment by natestemen** [comment]
> Output of mitiq.about() is `NoneType` not `str`.

You're right! I shouldn't have said "_... returns a string_".

Exactly as you say, the type annotations denote **return** types, without any indication if something is printed to standard out, standard error, or any other external action is taken. What I'm lightly suggesting (again, low priority) is we ensure that `mitiq.about()` prints to standard out.

File: mitiq/tests/test_about.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+# LICENSE file in the root directory of this source tree.
+
+"""Tests for mitiq.about()."""
+
+import mitiq
+
+
+def test_stdout():
+    """Tests function prints a str."""
+    mitiq.about()
```
Comment (natestemen) [inline mitiq/tests/test_about.py:13]: I don't think this tests that `mitiq.about()` prints to standard out (at least when run using `pytest mitiq/tests/test_about.py` which, IMO, we should support).

There is an example of capturing standard output in the pytest docs: https://docs.pytest.org/en/6.2.x/capture.html#accessing-captured-output-from-a-test-function

File: Makefile [hunk 2]
```diff
@@ -61,6 +61,7 @@ requirements: requirements/requirements.txt
 .PHONY: test
 test:
 	pytest -n auto -v --cov=mitiq --cov-report=term --cov-report=xml --ignore=mitiq/interface/mitiq_pyquil
+	pytest -rP mitiq/tests/test_about.py
```
Comment (natestemen) [inline Makefile:64]: Why is this needed? If I run the following command to list all the tests that will be run, `test_about.py::test_stdout` is included.
```bash
pytest --collect-only -n auto -v --cov=mitiq --cov-report=term --cov-report=xml --ignore=mitiq/interface/mitiq_pyquil

**Review by natestemen — CHANGES_REQUESTED** [review]
Thanks for cleaning this up! Looks like it's not quite working yet, but will be nice to have.

File: Makefile (hunk 2 above)
Comment (purva-thakre) [reply Makefile:64]: It's printing the output to verify it works. 

image

File: mitiq/tests/test_about.py (hunk 1 above)
Comment (purva-thakre) [reply mitiq/tests/test_about.py:13]: Yes, I did see that example before choosing to use `pytest -rP mitiq/tests/test_about.py`

I'll read more on the differences between capturing an stdout vs. testing an stdout.

File: Makefile (hunk 2 above)
Comment (natestemen) [reply Makefile:64]: I see that `-rP` means that extra info will be printed when tests pass **and** have output, but

1. I don't think the test is verifying there _is_ output,[^1] and
2. I don't think we need this to be printed every time `make test` is run. As long as pytest does in fact that the printing works, we don't need to see it :)

In fact won't `test_stdout` be run twice with this PR in the current state?

[^1]: I changed the `print` statement in `about()` to `return 0` and the test still passed.

File: Makefile [hunk 3]
```diff
@@ -61,7 +61,6 @@ requirements: requirements/requirements.txt
 .PHONY: test
 test:
 	pytest -n auto -v --cov=mitiq --cov-report=term --cov-report=xml --ignore=mitiq/interface/mitiq_pyquil
-
```
Comment (natestemen) [inline Makefile:64]: Let's leave the spacing between the makefile commands. It makes the file a touch easier to read.

**Review by natestemen — APPROVED** [review]
LGTM! Just one small change requested before merging. Thanks!

File: Makefile (hunk 3 above)
Comment (purva-thakre) [reply Makefile:64]: Thanks for catching this! I shouldn't have made the change.
